### PR TITLE
remove netty reference in DemoSpec

### PR DESCRIPTION
**Pull request by sbl**
Stumbled upon this while reading through the code.

Partially addresses #298

### CODE REVIEW

File: demo/src/test/scala/io/finch/demo/DemoSpec.scala [hunk 1]
```diff
@@ -81,31 +78,23 @@ class DemoSpec extends FlatSpec with Matchers {
...
 
   def POST(path: String, body: Option[String]): Request = {
-    val r = mkSecretHttpRequest(HttpMethod.POST, path)
+    val r = mkSecretHttpRequest(Post, path)
 
     body.foreach { b =>
-      val buf = ChannelBuffers.copiedBuffer(b, Charset.defaultCharset())
-      r.setContent(buf)
-      r.headers().add("content-length", buf.readableBytes())
+      val ba = ByteArray.Owned(b.getBytes("UTF-8"))
```
Comment (vkostyukov) [inline demo/src/test/scala/io/finch/demo/DemoSpec.scala]: We should be able to do `Buf.Utf8(b)`.

**Comment by vkostyukov** [comment]
Neat! :+1:

File: demo/src/test/scala/io/finch/demo/DemoSpec.scala (hunk 1 above)
Comment (sbl) [inline demo/src/test/scala/io/finch/demo/DemoSpec.scala]: Updated.

**Comment by vkostyukov** [comment]
Thanks @sbl! I'm merging this.

**Pull request merged**